src/configuration-form/special-dates.jsx

```diff
@@ -48,6 +59,45 @@ class SpecialDates extends React.Component {
 		this.setState( { date: '', value: '' } );
 	};
 
+	onFileLoad = ( event ) => {
+		try {
+			const jcalData = ICAL.parse( event.target.result );
+			const vcalendar = new ICAL.Component( jcalData );
+			const vevents = vcalendar.getAllSubcomponents( 'vevent' );
+			vevents.forEach( ( vevent ) => {
+				const dtstart = vevent.getFirstPropertyValue( 'dtstart' );
+				const date = dayjs( dtstart.toJSDate() );
+				if ( date.year() !== Number( this.props.year ) ) {
```

Comment:
Not sure if you're aware, but the `year` starts as a number and becomes a string when you update the input. I converted it to a number here, but you may want to fix that in the `onChange` callback for `year`.

Reply:
Thanks for catching that - I'll fix it in a follow-up commit 🙇‍♂️

Reply:
Done in https://github.com/klimeryk/recalendar.js/commit/7450bcb65b6e468a10b702108574ae41c934ad0f 🙇‍♂️